src/pamiq_core/console/cui.py

```diff
@@ -0,0 +1,141 @@
+            if self._fetch_status() == "offline":
+                print(f'Command "{cmd_name}" not executed. Can\'t connect AMI system.')
+                return False
+        # Perform command
+        return super().onecmd(line)
+
+    @override
+    def do_help(self, arg: str) -> None:
+        """Show all commands and details."""
+        # Store methods with the same docstring
+        # groups = {"Doscstring": ["command", "c"]}
+        groups: dict[str, list[str]] = {}
+        for attr in dir(self):
+            if attr.startswith("do_"):
+                cmd_name = attr[3:]
+                method = getattr(self, attr)
+                doc = method.__doc__ or ""
+                doc = doc
+                if doc not in groups:
+                    groups[doc] = []
+                groups[doc].append(cmd_name)
+        # Define showing order
+        _doc_order = [
+            "Show all commands and details.",
+            "Pause the AMI system.",
+            "Resume the AMI system.",
+            "Save a checkpoint.",
+            "Shutdown the AMI system.",
+            "Exit the console.",
+        ]
+        assert set(_doc_order) == set(groups.keys())
+        # Show details in the format f"{c}/{command} {Docstring}"
+        for doc in _doc_order:
+            cmds = "/".join(sorted(groups[doc], key=len))
+            print(f"{cmds:<11} {doc}")
+
+    def do_h(self, arg: str) -> None:
+        """Show all commands and details."""
+        self.do_help(arg)
+
+    def do_pause(self, arg: str) -> None:
+        """Pause the AMI system."""
+        response = requests.post(f"http://{self._host}:{self._port}/api/pause")
+        print(json.loads(response.text)["result"])
+
+    def do_p(self, arg: str) -> None:
+        """Pause the AMI system."""
+        return self.do_pause(arg)
+
+    def do_resume(self, arg: str) -> None:
+        """Resume the AMI system."""
+        response = requests.post(f"http://{self._host}:{self._port}/api/resume")
+        print(json.loads(response.text)["result"])
+
+    def do_r(self, arg: str) -> None:
+        """Resume the AMI system."""
+        return self.do_resume(arg)
+
+    def do_shutdown(self, arg: str) -> bool:
+        """Shutdown the AMI system."""
+        confirm = input("Confirm AMI system shutdown? (y/n): ")
+        if confirm.lower() in ["y", "yes"]:
+            response = requests.post(f"http://{self._host}:{self._port}/api/shutdown")
+            print(json.loads(response.text)["result"])
+            return True
+        print("Shutdown cancelled.")
+        return False
+
+    def do_s(self, arg: str) -> bool:
+        """Shutdown the AMI system."""
+        return self.do_shutdown(arg)
+
+    def do_quit(self, arg: str) -> bool:
+        """Exit the console."""
+        return True
+
+    def do_q(self, arg: str) -> bool:
+        """Exit the console."""
+        return self.do_quit(arg)
+
+    def do_ckpt(self, arg: str) -> None:
+        """Save a checkpoint."""
```

Comment:
pamiq-coreでは`ckpt`ではなく`state`に統一しているので、動作名としては `save-state`が適切。ただそれだとコマンドを打ちにくいので、`save`にすると良いと思う。

```suggestion
    def do_save(self, arg: str) -> None:
        """Save state to disk."""
```

Reply:
このまま`save`に変えると頭文字が`shutdown`とかぶってしまうのが気になる。となると`shutdown`を別なものに変えることになると思うんだけど、今度はそうすると`shutdown`の名称がpamiq-coreの内部でのものと一致しなくなるというもどかしさ
これどうしよう

Reply:
短縮形を持たないようにすると良い。`do_s`, `do_c`を消せば良い。

Reply:
`do_s`は`shutdown`から`save`の短縮系に変更、`do_c`は削除した https://github.com/MLShukai/pamiq-core/pull/210/commits/5152760f6cdb4efd9314c3de7273c73fd71b7a1e